### PR TITLE
[LFXV2-313, 318, 319, 320] Add Past Meeting Summaries Support with Email Notifications

### PR DESCRIPTION
**Pull request by andrest50**
## Summary
This PR implements a comprehensive past meeting summaries feature that allows meeting participants to view AI-generated summaries of completed meetings, with host approval workflows and automatic email notifications.

### 📝 Past Meeting Summaries API
- **GET** `/past_meetings/{past_meeting_uid}/summaries/{summary_uid}` - Retrieve specific summary
- **PUT** `/past_meetings/{past_meeting_uid}/summaries/{summary_uid}` - Update and approve summaries
- Complete CRUD operations with ETag support for optimistic concurrency control

### 🏗️ Data Infrastructure  
- New NATS KV bucket `past-meeting-summaries` for persistent storage
- Domain models with comprehensive tagging for search/indexing
- Repository pattern implementation with full CRUD support
- Integration with existing meeting and registrant systems

### 📧 Email Notification System
- Automatic email notifications to meeting hosts when summaries are available
- Professional HTML and text email templates with LFX branding
- Graceful error handling (email failures don't prevent summary operations)
- Integration with existing SMTP email infrastructure

### 🎯 Zoom Webhook Integration
- Process `meeting.summary_completed` webhook events
- Automatic summary creation from Zoom AI summaries
- Support for summary content, document URLs, and metadata
- Updated to use current Zoom API fields (removed deprecated overview/details/next_steps)

### ✅ Host Approval Workflow
- Summaries require host approval before being visible to participants
- Hosts can edit summary content before approval
- Email notifications sent automatically when summaries are ready for review
- Approval status tracking and audit trail

## API Schema Updates
- Modernized summary data structure to align with current Zoom API
- Replaced deprecated fields with `content`, `doc_url`, and `edited_content`
- Simplified update payload for better developer experience
- Updated all OpenAPI specifications and generated client code

### Database Schema
- Past meeting summaries stored in dedicated NATS KV bucket
- Relationships maintained with past meetings and original meetings
- Full audit trail with created/updated timestamps
- Platform-agnostic design supporting future meeting providers

### Security & Performance
- JWT authentication required for all endpoints  
- ETag-based optimistic concurrency control
- Efficient filtering for host registrants only
- Comprehensive input validation and error handling

### Testing & Quality
- Full unit test coverage for all domain models
- Mock implementations for all external dependencies
- Integration with existing test infrastructure
- Comprehensive error handling and logging

## Test Plan
- [x] Summary creation via Zoom webhooks
- [x] Host email notifications 
- [x] Summary retrieval and updates
- [x] Approval workflow
- [x] ETag concurrency control
- [x] Error handling and edge cases

## Manual Testing

Post request to `/webhooks/zoom` with a payload for the `meeting.summary_completed` event type:

```
{
  "event": "meeting.summary_completed",
  "event_ts": 1626230691572,
  "payload": {
    "account_id": "dzVA4QmMQfyISoRcpFO8CA",
    "object": {
      "meeting_host_id": "30R7kT7bTIKSNUFEuH_Qlg",
      "meeting_host_email": "jchill@example.com",
      "meeting_uuid": "aDYlohsHRtCd4ii1uC2+hA==",
      "meeting_id": 96620450802,
      "meeting_topic": "My Meeting",
      "meeting_start_time": "2025-09-15T15:00:00Z",
      "meeting_end_time": "2025-09-15T15:30:00Z",
      "summary_start_time": "2025-09-15T15:00:00Z",
      "summary_end_time": "2025-09-15T15:30:00Z",
      "summary_created_time": "2025-09-15T15:00:00Z",
      "summary_last_modified_time": "2025-09-15T15:00:00Z",
      "summary_title": "Meeting Summary for my meeting",
      "summary_overview": "Meeting Overview",
      "summary_details": [
        {
          "label": "Meeting Overview",
          "summary": "Meeting Overview"
        }
      ],
      "next_steps": [
        "step1"
      ],
      "summary_content": "## Key takeaways\n- Mobile app performance issues are affecting user retention.\n- New onboarding flow received positive feedback from beta testers.\n- Need to prioritize accessibility improvements.\n- Customer support response time has improved by 25%.\n\n## Discussed topics\n### Mobile App Performance\nDiscussion of recent performance metrics and user complaints\n- **Details**\n    - Sarah (Product): Reports of app crashes increased 15% this month\n    - Mike (Engineering): Memory optimization needed in latest release\n    - Tom (QA): Identified memory leak in photo upload feature\n- **Conclusion**\n    - Implement performance monitoring tools\n    - Prioritize memory optimization in next sprint\n\n### Onboarding Flow\nReview of beta testing results for new user onboarding\n- **Details**\n    - Rachel (UX): 90% completion rate in beta testing\n    - David (Product): Positive feedback on simplified registration\n- **Conclusion**\n    - Ready for full rollout next month\n    - Need to monitor analytics post-launch\n\n### Accessibility Compliance\nDiscussion of current accessibility status and needed improvements\n- **Details**\n    - Lisa (Design): Screen reader compatibility issues identified\n    - John (Engineering): WCAG compliance at 80%\n- **Conclusion**\n    - Create accessibility improvement roadmap\n    - Schedule external audit\n\n## Challenges\n* Resource constraints for performance optimization\n* Integration testing environment stability issues\n* Lack of accessibility expertise in the team\n\n## Action items\n- **Sarah**\n  - Prepare performance monitoring implementation plan\n  - Schedule follow-up meeting with engineering team\n- **Mike**\n  - Investigate memory leak fix\n  - Document performance optimization guidelines\n- **Lisa**\n  - Create accessibility improvement proposal\n  - Research accessibility testing tools\n- **Rachel**\n  - Prepare onboarding analytics dashboard\n  - Document beta testing findings",
      "summary_doc_url": "https://docs.example.com/doc/1aBcDeFgHiJkLmNoPqRsTu"
    }
  }
}
```

Response:

```
200 OK
{
    "status": "success",
    "message": "Event meeting.summary_completed queued for processing"
}
```

List past meeting summaries endpoint after summary is inserted from webhook event:

```
200 OK
{
    "summaries": [
        {
            "uid": "75b8c214-42b4-4ade-af34-9cb3ff05a9a3",
            "past_meeting_uid": "d489fcc8-0a23-4449-bcc7-76a0118bc450",
            "meeting_uid": "6fe7dde3-e761-4aa7-9152-a41a8082f70a",
            "platform": "Zoom",
            "password": "eb05d100-a927-45a4-8df4-eb01878a15ed",
            "zoom_config": {
                "meeting_id": "96620450802",
                "meeting_uuid": "aDYlohsHRtCd4ii1uC2+hA=="
            },
            "summary_data": {
                "start_time": "2025-09-15T15:00:00Z",
                "end_time": "2025-09-15T15:30:00Z",
                "title": "Meeting Summary for my meeting",
                "content": "## Key takeaways\n- Mobile app performance issues are affecting user retention.\n- New onboarding flow received positive feedback from beta testers.\n- Need to prioritize accessibility improvements.\n- Customer support response time has improved by 25%.\n\n## Discussed topics\n### Mobile App Performance\nDiscussion of recent performance metrics and user complaints\n- **Details**\n    - Sarah (Product): Reports of app crashes increased 15% this month\n    - Mike (Engineering): Memory optimization needed in latest release\n    - Tom (QA): Identified memory leak in photo upload feature\n- **Conclusion**\n    - Implement performance monitoring tools\n    - Prioritize memory optimization in next sprint\n\n### Onboarding Flow\nReview of beta testing results for new user onboarding\n- **Details**\n    - Rachel (UX): 90% completion rate in beta testing\n    - David (Product): Positive feedback on simplified registration\n- **Conclusion**\n    - Ready for full rollout next month\n    - Need to monitor analytics post-launch\n\n### Accessibility Compliance\nDiscussion of current accessibility status and needed improvements\n- **Details**\n    - Lisa (Design): Screen reader compatibility issues identified\n    - John (Engineering): WCAG compliance at 80%\n- **Conclusion**\n    - Create accessibility improvement roadmap\n    - Schedule external audit\n\n## Challenges\n* Resource constraints for performance optimization\n* Integration testing environment stability issues\n* Lack of accessibility expertise in the team\n\n## Action items\n- **Sarah**\n  - Prepare performance monitoring implementation plan\n  - Schedule follow-up meeting with engineering team\n- **Mike**\n  - Investigate memory leak fix\n  - Document performance optimization guidelines\n- **Lisa**\n  - Create accessibility improvement proposal\n  - Research accessibility testing tools\n- **Rachel**\n  - Prepare onboarding analytics dashboard\n  - Document beta testing findings",
                "doc_url": "https://docs.example.com/doc/1aBcDeFgHiJkLmNoPqRsTu",
                "edited_content": ""
            },
            "requires_approval": true,
            "approved": false,
            "email_sent": true,
            "created_at": "2025-09-12T00:38:37Z",
            "updated_at": "2025-09-12T00:38:37Z"
        }
    ]
}
```

Get past meeting summary endpoint returns the created summary:

```
200 OK
{
    "uid": "75b8c214-42b4-4ade-af34-9cb3ff05a9a3",
    "past_meeting_uid": "d489fcc8-0a23-4449-bcc7-76a0118bc450",
    "meeting_uid": "6fe7dde3-e761-4aa7-9152-a41a8082f70a",
    "platform": "Zoom",
    "password": "eb05d100-a927-45a4-8df4-eb01878a15ed",
    "zoom_config": {
        "meeting_id": "96620450802",
        "meeting_uuid": "aDYlohsHRtCd4ii1uC2+hA=="
    },
    "summary_data": {
        "start_time": "2025-09-15T15:00:00Z",
        "end_time": "2025-09-15T15:30:00Z",
        "title": "Meeting Summary for my meeting",
        "content": "## Key takeaways\n- Mobile app performance issues are affecting user retention.\n- New onboarding flow received positive feedback from beta testers.\n- Need to prioritize accessibility improvements.\n- Customer support response time has improved by 25%.\n\n## Discussed topics\n### Mobile App Performance\nDiscussion of recent performance metrics and user complaints\n- **Details**\n    - Sarah (Product): Reports of app crashes increased 15% this month\n    - Mike (Engineering): Memory optimization needed in latest release\n    - Tom (QA): Identified memory leak in photo upload feature\n- **Conclusion**\n    - Implement performance monitoring tools\n    - Prioritize memory optimization in next sprint\n\n### Onboarding Flow\nReview of beta testing results for new user onboarding\n- **Details**\n    - Rachel (UX): 90% completion rate in beta testing\n    - David (Product): Positive feedback on simplified registration\n- **Conclusion**\n    - Ready for full rollout next month\n    - Need to monitor analytics post-launch\n\n### Accessibility Compliance\nDiscussion of current accessibility status and needed improvements\n- **Details**\n    - Lisa (Design): Screen reader compatibility issues identified\n    - John (Engineering): WCAG compliance at 80%\n- **Conclusion**\n    - Create accessibility improvement roadmap\n    - Schedule external audit\n\n## Challenges\n* Resource constraints for performance optimization\n* Integration testing environment stability issues\n* Lack of accessibility expertise in the team\n\n## Action items\n- **Sarah**\n  - Prepare performance monitoring implementation plan\n  - Schedule follow-up meeting with engineering team\n- **Mike**\n  - Investigate memory leak fix\n  - Document performance optimization guidelines\n- **Lisa**\n  - Create accessibility improvement proposal\n  - Research accessibility testing tools\n- **Rachel**\n  - Prepare onboarding analytics dashboard\n  - Document beta testing findings",
        "doc_url": "https://docs.example.com/doc/1aBcDeFgHiJkLmNoPqRsTu",
        "edited_content": ""
    },
    "requires_approval": true,
    "approved": false,
    "email_sent": true,
    "created_at": "2025-09-12T00:38:37Z",
    "updated_at": "2025-09-12T00:38:37Z"
}
```

Update past meeting summary endpoint request:

```
{
    "uid": "e3e87dd7-e565-4341-bfee-c5b310176bee",
    "past_meeting_uid": "d489fcc8-0a23-4449-bcc7-76a0118bc450",
    "meeting_uid": "6fe7dde3-e761-4aa7-9152-a41a8082f70a",
    "edited_content": "updated",
    "approved": true
}
```

Response:

```
200 OK
{
    "uid": "75b8c214-42b4-4ade-af34-9cb3ff05a9a3",
    "past_meeting_uid": "d489fcc8-0a23-4449-bcc7-76a0118bc450",
    "meeting_uid": "6fe7dde3-e761-4aa7-9152-a41a8082f70a",
    "platform": "Zoom",
    "password": "eb05d100-a927-45a4-8df4-eb01878a15ed",
    "zoom_config": {
        "meeting_id": "96620450802",
        "meeting_uuid": "aDYlohsHRtCd4ii1uC2+hA=="
    },
    "summary_data": {
        "start_time": "2025-09-15T15:00:00Z",
        "end_time": "2025-09-15T15:30:00Z",
        "title": "Meeting Summary for my meeting",
        "content": "## Key takeaways\n- Mobile app performance issues are affecting user retention.\n- New onboarding flow received positive feedback from beta testers.\n- Need to prioritize accessibility improvements.\n- Customer support response time has improved by 25%.\n\n## Discussed topics\n### Mobile App Performance\nDiscussion of recent performance metrics and user complaints\n- **Details**\n    - Sarah (Product): Reports of app crashes increased 15% this month\n    - Mike (Engineering): Memory optimization needed in latest release\n    - Tom (QA): Identified memory leak in photo upload feature\n- **Conclusion**\n    - Implement performance monitoring tools\n    - Prioritize memory optimization in next sprint\n\n### Onboarding Flow\nReview of beta testing results for new user onboarding\n- **Details**\n    - Rachel (UX): 90% completion rate in beta testing\n    - David (Product): Positive feedback on simplified registration\n- **Conclusion**\n    - Ready for full rollout next month\n    - Need to monitor analytics post-launch\n\n### Accessibility Compliance\nDiscussion of current accessibility status and needed improvements\n- **Details**\n    - Lisa (Design): Screen reader compatibility issues identified\n    - John (Engineering): WCAG compliance at 80%\n- **Conclusion**\n    - Create accessibility improvement roadmap\n    - Schedule external audit\n\n## Challenges\n* Resource constraints for performance optimization\n* Integration testing environment stability issues\n* Lack of accessibility expertise in the team\n\n## Action items\n- **Sarah**\n  - Prepare performance monitoring implementation plan\n  - Schedule follow-up meeting with engineering team\n- **Mike**\n  - Investigate memory leak fix\n  - Document performance optimization guidelines\n- **Lisa**\n  - Create accessibility improvement proposal\n  - Research accessibility testing tools\n- **Rachel**\n  - Prepare onboarding analytics dashboard\n  - Document beta testing findings",
        "doc_url": "https://docs.example.com/doc/1aBcDeFgHiJkLmNoPqRsTu",
        "edited_content": "updated"
    },
    "requires_approval": true,
    "approved": true,
    "email_sent": true,
    "created_at": "2025-09-12T00:38:37Z",
    "updated_at": "2025-09-12T00:45:04Z"
}
```

## Related Issues
- LFXV2-313: Past Meeting Summary API endpoints
- LFXV2-318: Host approval workflow
- LFXV2-319: Email notifications for summaries
- LFXV2-320: Zoom webhook integration for summaries

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: cmd/meeting-api/api_past_meeting_summaries.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	updateSummaryReq := service.ConvertUpdatePastMeetingSummaryPayloadToDomain(payload)
+
+	summary, err := s.pastMeetingSummaryService.UpdateSummary(ctx, updateSummaryReq, etag)
+	if err != nil {
+		return nil, handleError(err)
+	}
+
+	return service.ConvertDomainToPastMeetingSummaryResponse(summary), nil
```
[the remark this answers is not in the document]
Comment (andrest50) [reply cmd/meeting-api/api_past_meeting_summaries.go:74]: It's a PUT endpoint so you must provide the fields upon every update. So if there is no approved field, you are essentially set it to false. Same regarding the edited content.

**Review by jordane — APPROVED** [review]
